### ISSUE TITLE
Should try symlink templates files when files are pointing to absolute well known directories.

### ISSUE THREAD

**Issue by 3v1n0** [twpayne/chezmoi#1075]
## Is your feature request related to a problem? Please describe.

This is a bit related to https://github.com/twpayne/chezmoi/issues/167#issuecomment-802005513 so, when a symlink is added to chezmoi, it should instead try to create a symlink template if the symlink is pointing to a path in `{{ .chezmoi.homedir }}` or `{{ .chezmoi.sourceDir }}`  (it's actually the main purpose of my scripts as documented in https://github.com/twpayne/chezmoi/issues/726#issuecomment-636518384)

This will make such links to be more portable, without having to care at local level to use relative paths.

## Describe the solution you'd like

When a symlink is detected:
 - If it's an absolute path and it's pointing to a sub-dir of the home or chezmoi dir, it should create a symlink template instead.

## Describe alternatives you've considered

Basically having scripts doing this for me

**Comment by twpayne** [comment]
Thanks to @Flygrounder for implementing this in #1124. See #1139 for a couple of minor tweaks.
